pintc/src/predicate/analyse/type_check/custom_types.rs

```diff
@@ -113,118 +222,123 @@ impl Contract {
-                    .find_map(|(_, fld_ty)| get_custom_type_mut_ref(custom_path, fld_ty)),
-                Type::Custom { path, .. } => (path == custom_path).then_some(ty),
-                Type::Alias { ty, .. } => get_custom_type_mut_ref(custom_path, ty),
-                Type::Map { ty_from, ty_to, .. } => get_custom_type_mut_ref(custom_path, ty_from)
-                    .or_else(|| get_custom_type_mut_ref(custom_path, ty_to)),
-                Type::Vector { ty, .. } => get_custom_type_mut_ref(custom_path, ty),
-                Type::Error(_)
-                | Type::Unknown(_)
-                | Type::Any(_)
-                | Type::Primitive { .. }
-                | Type::Union { .. } => None,
-            }
-        }
-
-        // Any custom types referred to *within new types* need to be converted to type aliases.
-        // E.g.,
-        //   type A = int;
-        //   type B = { A, A };
-        //   // B will have `Type::Custom("A")` which need to be `Type::Alias("A", int)`
-
-        for new_type_idx in 0..self.new_types.len() {
-            // We're replacing only a single new type at a time, if found in other new-types.
-            let new_type = self.new_types[new_type_idx].clone();
-
-            // Replace the next found custom type which needs to be replaced with a an alias.
-            // There may be multiple replacements required per iteration, so we'll visit every
-            // current new-type decl per iteration until none are updated.
-            for loop_check in 0.. {
-                let mut modified = false;
-
-                for NewTypeDecl { ref mut ty, .. } in &mut self.new_types {
-                    if let Some(custom_ty) = get_custom_type_mut_ref(&new_type.name.name, ty) {
-                        *custom_ty = Type::Alias {
-                            path: new_type.name.name.clone(),
-                            ty: Box::new(new_type.ty.clone()),
-                            span: new_type.span.clone(),
-                        };
-
-                        modified = true;
-                    }
-                }
-
-                if !modified {
-                    break;
-                }
-
-                if loop_check > 10_000 {
-                    return Err(handler.emit_err(Error::Compile {
-                        error: CompileError::Internal {
-                            msg: "infinite loop in lower_custom_types_in_newtypes()",
-                            span: empty_span(),
-                        },
-                    }));
-                }
-            }
-        }
-
-        Ok(())
-    }
-
-    /// Lower every `Type::Custom` type in a contract to a `Type::Alias` type if the custom type
-    /// actually matches one of the new type declarations in the contract. All remaining custom
-    /// types after this method runs should refer to enums.
+    //fn lower_custom_types_in_newtypes(&mut self, handler: &Handler) -> Result<(), ErrorEmitted> {
```

Comment:
Are we going to need this in the future?

Reply:
Ooh, I wasn't sure at the time, and then I forgot about it, and now all the tests pass, so I guess not.